src/commands.ts

```diff
@@ -22,9 +23,34 @@ export function init(context: ExtensionContext) {
     commands.registerCommand(
       'diffMerge.compareFileWithClipboard',
       compareFileWithClipboard
-    )
+    ),
+    commands.registerCommand('diffMerge.openWithDiffMerge', reopenCurrentWithDiffMerge),
   );
 
+  async function reopenCurrentWithDiffMerge() {
+    const editor = window.activeTextEditor;
+    if (!editor) return;
+    const visibleEditors = window.visibleTextEditors;
+    const editorIndex = visibleEditors.indexOf(editor);
+    // This might not be always true. But it's the only way I can think of how to know whether it's a diff-editor.
+    const isDiffEditor = (editor: TextEditor) => editor.viewColumn === undefined && ['git', 'file'].includes(editor.document.uri.scheme);
+    // In case multiple diff-editors are open, we need to find the correct one.
+    const diffEditors =
+      takeRightWhile(visibleEditors.slice(undefined, editorIndex + 1), isDiffEditor)
+        .concat(takeWhile(visibleEditors.slice(editorIndex + 1), isDiffEditor));
+    const editorIndexInDiffEditors = diffEditors.indexOf(editor);
+    // Get the editors content.
+    const leftEditorIndex = Math.trunc(editorIndexInDiffEditors / 2) * 2;
+    const rightEditorIndex = leftEditorIndex + 1;
+    const leftEditor = diffEditors[leftEditorIndex];
+    const leftContent = leftEditor.document.getText();
+    const leftPath = leftEditor.document.uri.fsPath;
+    const rightEditor = diffEditors[rightEditorIndex];
```

Comment:
I'm probably missing something.. If `diffEditors` contains only the 2 editors, the first is the left and the second is the right, isn't?

```suggestion
    const [leftEditor, rightEditor] = diffEditors;
    const leftContent = leftEditor.document.getText();
    const leftPath = leftEditor.document.uri.fsPath;
```

Reply:
no, look at the comment above. there's might be a case that diff editors can be 4 or 6, if multiple diff are open and consecutive.
but if you'd like, you can actually refactor it to use `slice(leftIndex, rightIndex+1)` and apply the same logic you suggested.

Reply:
see latest commit, addresses this.